web/components/views/view-list-header.tsx

```diff
@@ -0,0 +1,84 @@
+import React, { useRef, useState } from "react";
+import { observer } from "mobx-react";
+// icons
+import { Search, X } from "lucide-react";
+// helpers
+import { cn } from "@/helpers/common.helper";
+// hooks
+import { useProjectView } from "@/hooks/store";
+import useOutsideClickDetector from "@/hooks/use-outside-click-detector";
+
+export const ViewListHeader = observer(() => {
+  // states
+  const [isSearchOpen, setIsSearchOpen] = useState(false);
+  // const [isSearchOpen, setIsSearchOpen] = useState(searchQuery !== "" ? true : false);
+  // refs
+  const inputRef = useRef<HTMLInputElement>(null);
+
+  const { searchQuery, updateSearchQuery } = useProjectView();
+
+  // handlers
+  const handleInputKeyDown = (e: React.KeyboardEvent<HTMLInputElement>) => {
+    if (e.key === "Escape") {
+      if (searchQuery && searchQuery.trim() !== "") updateSearchQuery("");
+      else {
+        setIsSearchOpen(false);
+        inputRef.current?.blur();
+      }
+    }
+  };
+
+  // outside click detector hook
+  useOutsideClickDetector(inputRef, () => {
+    if (isSearchOpen && searchQuery.trim() === "") setIsSearchOpen(false);
+  });
+
+  return (
```

Comment:
This search should be extracted out into a component

https://app.plane.so/plane/projects/02c3e1d5-d7e2-401d-a773-45ecba45d745/issues/0c372512-8c77-4648-b322-aa55ed0a6dd7